pkg/detector/detector.go

```diff
@@ -1021,7 +1025,7 @@ func (d *ResourceDetector) ReconcilePropagationPolicy(key util.QueueKey) error {
 	if err != nil {
 		if apierrors.IsNotFound(err) {
 			klog.Infof("PropagationPolicy(%s) has been removed.", ckey.NamespaceKey())
-			return d.HandlePropagationPolicyDeletion(ckey.Namespace, ckey.Name)
+			return d.HandlePropagationPolicyDeletion(ckey.PermanentID)
```

Comment:
Have you ever considered adding a finalizer to PropagationPolicy/ClusterPropagationPolicy, and doing the cleanup work before it has been removed? So that we can get the permanent ID from the object, instead of building a special key for it.

Reply:
agree with it.

Reply:
Speaking of the finalizer, I haven't thought about it thoroughly. Just feel it's the right pattern to deal with resource cleanup.

Reply:
Yes, it's the right pattern to deal with resource cleanup.I would like to explain my concerns about the current pattern: 
**If all instances of the karmada-controller-manager suddenly crash when the policy is deleted, the deletion events will be lost even if the karmada-controller-manager returns to normal.**

 To be honest, we magically modified this part of the code to achieve this function. It's like the following yaml.
![image](https://github.com/karmada-io/karmada/assets/89241565/0e4ac199-5a5f-4470-bf5d-fd30ad7086a3)

As I said above, I strongly agree with this idea, but I don't know whether it is suitable for the community or whether it breaks the usage habits of other users.

Reply:
> I would like to explain my concerns about the current pattern:
If all instances of the karmada-controller-manager suddenly crash when the policy is deleted, the deletion events will be lost even if the karmada-controller-manager returns to normal.

+100
Thanks.

Reply:
> As I said above, I strongly agree with this idea, but I don't know whether it is suitable for the community or whether it breaks the usage habits of other users.

Maybe I don't get your point. I think users don't have to care about the finalizer, so basically non-user-facing.

Reply:
> I think users don't have to care about the finalizer, so basically non-user-facing.

I'm not sure if such a case exists. For example, if a developer uses the PropagationPolicy in secondary development, will the finalizer affect his judgment on the deletion logic?

Reply:
I don't think so. no worries.

Reply:
so, are you going to send another PR for the finalizer thing?

Reply:
Now that I got your affirmative reply, I think I'll do it. To avoid conflicts, I will submit it after this PR.

Reply:
why after this PR? If after this PR, we have to revert part of this PR.